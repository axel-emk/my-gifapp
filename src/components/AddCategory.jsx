import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState([ '']);

    const onInputChange = ( {target} ) => {
        // console.log( target.value );
        setInputValue( target.value );
    }

    const onSubmit = ( event ) => {
        // console.log('Hola mundo desde Submit');
        event.preventDefault();        
        if( inputValue.trim().length <= 1) return;
        // console.log(inputValue);
        // setCategories( categories => [ inputValue, ...categories ]);
        onNewCategory(inputValue.trim());
        setInputValue('');


    }

    return (
        <>
              

        
            <form onSubmit={onSubmit} aria-label="form">
                <div className="input-group mb-3">
                    <button onClick={onSubmit}  className="btn btn-outline-secondary" type="button" id="button-addon1">Buscar</button>
                    <input className="input-group flex-nowrap form-control"
                    type="text"
                    placeholder="Buscar Gifs"
                    value={inputValue}
                    onChange={onInputChange} aria-label="Example text with button addon" aria-describedby="button-addon1"></input>
                </div>
            </form>


        {/*  */}
            {/* <form onSubmit={onSubmit} aria-label="form">
                <input className="input-group flex-nowrap form-control"
                    type="text"
                    placeholder="Buscar Gifs"
                    value={inputValue}
                    onChange={onInputChange}
                />


            </form> */}

        </>
    )
}

// TEst Proptypes - Esperar a una Function
AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}