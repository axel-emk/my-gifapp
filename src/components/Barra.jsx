


export const Barra = () => {
    return (
        <> 

        
        <form action="">
            <div className="input-group mb-3">
                <button className="btn btn-outline-secondary" type="button" id="button-addon1">Search</button>
                <input type="text" className="form-control" placeholder="" aria-label="Example text with button addon" aria-describedby="button-addon1"></input>
            </div>
            </form>
        </>
    )
}


// <form onSubmit={ onSubmit } aria-label="form">
// <input className="input-group flex-nowrap form-control"
// type="text" 
// placeholder="Buscar Gifs"
// value={ inputValue }
// onChange= { onInputChange  }
// />