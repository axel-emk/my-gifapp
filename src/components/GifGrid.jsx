import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs( category );

    // console.log({ isLoading});
    

    return (
        <>
            <h3>{ category }</h3>

            {/* Loading */}
            {
                isLoading && (<h2>Cargando...</h2>)  
            }
            

            <div className="card-grid">
                {
                    images.map( ( image ) => (
                        // <li key={ id }>{ title }</li>
                        <GifItem 
                             key={image.id}
                             { ...image }
                            //  title={ image.title }
                            //  url={ image.url }
                        />
                    ))
                }

            </div>


        </>
    )
}