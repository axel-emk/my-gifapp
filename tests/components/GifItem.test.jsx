import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Eren Jeger'
    const url = 'https://erenjeager.com'


    test('Debe de hacer match con el snapshot', () => {
        
        const { container } = render( <GifItem title={ title } url={ url } /> );
        expect ( container ).toMatchSnapshot();
    });
    
});