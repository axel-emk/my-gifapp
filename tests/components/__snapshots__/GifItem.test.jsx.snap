// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Eren Jeger"
      src="https://erenjeager.com/titan.jpg"
    />
    <p>
      Eren Jeger
    </p>
  </div>
</div>
`;
